components/CTA.tsx

```diff
@@ -65,16 +65,18 @@ export default function CTA({ balance }: { balance?: string }) {
   const { ctas } = useUserStore();
   const { openModal } = useContext(ModalContext);
 
-  const gloBalance = balance || 1000;
-  const totalYield = getTotalYield(Number(gloBalance));
+  const gloBalance = Number(balance) || 1000;
+  const totalYield = getTotalYield(gloBalance);
   const item = getImpactItems(totalYield)[0];
   const icons = item
     ? `${item.emoji} x ${item.count} ${item.description}`
     : "?";
 
   const email = Cookies.get("glo-email") || "";
 
-  const shareImpactText = `I just bought $${gloBalance} @glodollar.\n\nAt scale, this gives someone in extreme poverty enough money to buy ${icons} per year. Without me donating anything.\n\nLet’s end extreme poverty.`;
+  const shareImpactText = `I just bought $${gloBalance.toFixed(
```

Comment:
You can use https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/NumberFormat to format balance nicely too.